NYPL-discovery/discovery-front-end
# SCC-3172: Reservoir v1 update

**What's this do?**
Fixes JIRA Ticket [SCC-3172](https://jira.nypl.org/browse/SCC-3172).

This updates the version of the NYPL Reservoir Design System to version 1.0.5. Updates include:
- Removing enum types `LinkTypes`, `ButtonTypes`, and `IconNames`.
- Update props for the `Breadcrumbs`, `Button`, `Checkbox`, `Heading`, `HelperErrorText`, `Icon`, `Label`, `Link`, `SearchBar`, `Select`, and `SkeletonLoader` components.
- Updates the `Input` component implementation to `TextInput`.
- Refactors form-related component combination for the `Label` component, `<input>` HTML element, and `HelperErrorText` component.
- Updates layout styles from previous DS class rules.
- Tweaks style rules so that pages render appropriately.
- Updates unit tests accordingly.
- Removes the "nypl-ds" class name.
- Removes the `Modal` component but replaces it with the old `Modal` component implementation CSS for the `CancelConfirmationModal.jsx`, `ItemFiltersMobile.jsx`, and `TimedLogoutModal.jsx` components.
- Updated all CSS variable names to have the `--nypl` prefix.

**Why are we doing this? (w/ JIRA link if applicable)**
Updating the DS version will help create features faster and it'll keep the app up-to-date.

**Do these changes have automated tests?**
Yes. All unit tests passed and the tests related to DS component were updated accordingly.

**How should this be QAed?**
All pages need to be reviewed for:
- visual differences - these will be minor and subtle, nothing that affects how the app is used
- functional differences - there should be none, the app should work as expected, including all the forms.

~NOTE: Haven't verified SHEP-related pages just yet (waiting on VPN access).~ Update: I now have VPN access and can see the SHEP-related pages. There are no visual changes and everything works as expected.

Here are a few examples of pages side-by-side. On the left is the updated Reservoir and on the right is the production view.

## Home

<img width="1776" alt="Screen Shot 2022-07-20 at 3 25 54 PM" src="https://user-images.githubusercontent.com/1280564/180066394-203fc404-1ef9-47e7-b2f9-5460ade6dc20.png">

## Feedback

<img width="1227" alt="Screen Shot 2022-07-20 at 3 26 05 PM" src="https://user-images.githubusercontent.com/1280564/180066428-89ca76a1-7041-4848-a480-b37937c45e57.png">

## Search Results Page
<img width="1767" alt="Screen Shot 2022-07-20 at 3 26 33 PM" src="https://user-images.githubusercontent.com/1280564/180066489-9e2af1bd-d3e4-4eaa-8af7-11d0f2c9f5c5.png">

## Bib page
 
<img width="1763" alt="Screen Shot 2022-07-20 at 3 26 42 PM" src="https://user-images.githubusercontent.com/1280564/180066545-b42462d1-de64-45a7-9658-66ff449685f3.png">

## Mobile Filters

<img width="1109" alt="Screen Shot 2022-07-20 at 3 27 06 PM" src="https://user-images.githubusercontent.com/1280564/180066646-003d9904-8ba6-409b-a687-2bfdb83aae0b.png">

## Account Page

<img width="1762" alt="Screen Shot 2022-07-20 at 3 28 29 PM" src="https://user-images.githubusercontent.com/1280564/180066709-78d55e97-82f5-495e-8319-e9c50eb05200.png">

## Modal example

<img width="1978" alt="Screen Shot 2022-07-20 at 3 31 16 PM" src="https://user-images.githubusercontent.com/1280564/180066799-89ba4110-290c-4f34-af99-928faf0cf6c2.png">

## Advance Search

<img width="1999" alt="Screen Shot 2022-07-20 at 3 37 14 PM" src="https://user-images.githubusercontent.com/1280564/180067334-d75575f2-add7-4b56-8cd4-5b83c4747009.png">


**Dependencies for merging? Releasing to production?**
N/A

**Has the application documentation been updated for these changes?**
Yes

**Did someone actually run this code to verify it works?**
Yes, locally, including by logging in.
